plugins/ingest/src/main/java/org/elasticsearch/ingest/Data.java

```diff
@@ -129,4 +129,33 @@ public String getId() {
     public boolean isModified() {
         return modified;
     }
+
+    public Map<String, Object> asMap() {
```

Comment:
maybe Data should implement toXContent instead, given that this method is used only to output xcontent stuff?